### PR TITLE
Feat: Converted the main EP to typescript.

### PR DESCRIPTION
**Pull request by ITninja04**
Extended the base tsconfig to cover the new main.ts entry point.

Added a new build:desktop script to build the main.ts entry point to JS.

Added the aforementioned script to the dev and build scripts so the file is rebuilt before the app is launched.

### CODE REVIEW

**Review by itsMapleLeaf — CHANGES_REQUESTED** [review]
Thanks for this, this is a good change. Need to address a few things before merging this in. Also, make sure you run the `lint` and `format` scripts after you're done.

File: template/desktop/main.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { initRemix } from "remix-electron";
+import { app, BrowserWindow, dialog } from "electron";
+import { join } from "node:path";
+
+/** @type {BrowserWindow | undefined} */
```
[the remark this answers is not in the document]
Comment (ITninja04) [reply template/desktop/main.ts:5]: Sorry about those, I meant to clean them up and forgot about it. Will be pushing up changes in a few moments.

File: template/tsconfig.electron-build.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "noEmit": false,
+    "allowJs": true,
+    "checkJs": true,
+    "skipLibCheck": true
+  },
+  "include": [
+    "desktop/main.ts"
+  ],
+  "exclude": ["desktop/build", "public/build", ".cache"]
+}
```
Comment (cliffordfajardo) [inline template/tsconfig.electron-build.json:21]: Hi @itsMapleLeaf 
Given that electron bundles a full version of a relatively modern version of chrome, I believe its safe to make these changes, which you significantly reduce the bundle size and app speed

- change `target to -> `target: esnext`
   - docs: https://www.typescriptlang.org/tsconfig#target
- change `lib` to: `lib: ["DOM", "DOM.Iterable", "esnext"]`
    - this way we get modern types for the browser environment and node
    - docs: https://www.typescriptlang.org/tsconfig#lib

**Review by cliffordfajardo — CHANGES_REQUESTED** [review]

File: template/package.json [hunk 3]
```diff
@@ -6,11 +6,12 @@
...
     "prepare": "remix setup node",
-    "clean": "del-cli dist desktop/build public/build .cache",
-    "dev": "npm run clean && cross-env NODE_ENV=development npm-run-all --parallel --print-label --race dev:*",
+    "clean": "del-cli dist desktop/build public/build .cache desktop/main.js",
+    "dev": "npm run clean && npm run build:desktop && cross-env NODE_ENV=development npm-run-all --parallel --print-label --race dev:*",
     "dev:remix": "remix watch",
     "dev:nodemon": "wait-on file:desktop/main.js && nodemon .",
-    "build": "npm run clean && remix build && electron-builder",
+    "build": "npm run clean && npm run build:desktop && remix build && electron-builder",
+    "build:desktop": "yarn tsc -p tsconfig.electron-build.json",
```
[the remark this answers is not in the document]
Comment (cliffordfajardo) [reply template/package.json:14]: Also noticed we're on typescript 4.5.5, current the [latest stable typescript version](https://www.npmjs.com/package/typescript) is `4.6.2`

Should we update it in this PR?

**Comment by cliffordfajardo** [comment]
@ITninja04 the README.md file inside of the template file could use some minor updates for first time users😄 
- https://gist.github.com/cliffordfajardo/4fa9348f4711510357ebb90dd881d50d

File: template/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (itsMapleLeaf) [reply template/package.json:14]: Sure, fine by me

File: template/tsconfig.electron-build.json (hunk 2 above)
Comment (itsMapleLeaf) [reply template/tsconfig.electron-build.json:21]: I think the rationale is a good one, and I'm pretty much always in favor of targeting modern JS, even outside of "stable" environments like Electron. But [here](https://github.com/itsMapleLeaf/remix-electron/pull/11#discussion_r818947752), I suggested to use tsup instead, which uses esbuild under the hood, which should accomplish the same goal, and changing the TSConfig target doesn't affect the emit.

Changing the lib is fine though

**Comment by itsMapleLeaf** [comment]
> @ITninja04 the README.md file inside of the template file could use some minor updates for first time userssmile
> 
> * [gist.github.com/cliffordfajardo/4fa9348f4711510357ebb90dd881d50d](https://gist.github.com/cliffordfajardo/4fa9348f4711510357ebb90dd881d50d)

These are good changes, and I'll probably incorporate those outside of this pr, thanks :slightly_smiling_face: Although I don't recommend using yarn for public documentation, as less people are familiar with it, + yarn/pnpm users will likely be aware of the analogs from npm.

**Pull request closed without merging**